Add theme switcher

LiAuTraver opened the pull request:
Add light, dark, and follow the system theme option. Default to follow the System theme, user can change by toggling the settings(see pic below), and the theme will be remembered.
Worked fine with `Auto Dark Mode`.


Related feature request:
https://github.com/unchihugo/FluentFlyout/issues/6#issue-2629787980


![image](https://github.com/user-attachments/assets/876df210-a0d2-4c2e-9ea6-5028c1edbab3)

*p.s. Mainly not a c sharp dev, but love your app and hope this can help the app get better!* 😊

LiAuTraver commented:
Sorry for the formatting issues in the git diff.  did not find it untill now

unchihugo commented on FluentFlyoutWPF/App.xaml:
```diff
@@ -9,12 +9,11 @@
     <Application.Resources>
         <ResourceDictionary>
             <ResourceDictionary.MergedDictionaries>
-                <ui:ThemesDictionary Theme="Light" />
```
Adding `<ui:ThemesDictionary Theme="Unknown"/>` will fix some issues related to theming

unchihugo commented on FluentFlyoutWPF/Classes/ThemeManager.cs:
```diff
@@ -0,0 +1,78 @@
+    /// Applies the specified theme. See also <see href="https://github.com/Simnico99/MicaWPF/wiki/Change-Theme-or-Accent-color"/>.
+    /// </summary>
+    /// <param name="theme">The theme to apply. 1 for Light, 2 for Dark, 0 or any other value for System Default.</param>
+    private static void applyTheme(int theme)
```
This function works mostly well so far, but it should also update the MicaWPF brushes

unchihugo reviewed (changes requested):
Hi, I really appreciate what you've done! 

It seems like the theme switcher is working well overall, but since I used both MicaWPF and WPFui, there's still some areas where the theme isn't fully switching. I found out that adding `<ui:ThemesDictionary Theme="Unknown"/>` to the ResourceDictionary fixes some of those issues. However, the MicaWPF Brushes aren’t switching to their theme-specific counterparts.

Could you take a look at this before I merge the PR? Thank you!

LiAuTraver commented:
it's been some messy here, i forgot to pull from main branch before my commit.

overall mainly changes are:
- added theme switch
- repaired issues related with accent colors and brush 
- refactored the namespaces according to their relative path and code readability (from `FluentFlyoutWPF` to `FluentFlyout`) *actually I am not willing to change the namespace, but the annoying Resharper changed the naming automatically :-/ Since its a good practice, let's keep it. Reject if not desired* 😂
- specify the `WindowsMediaController.MediaManager` because you have a class with the same name
- The text in ComboBox was too left-aligned, so I added a space before every text in combo boxes.

unchihugo commented:
It looks like the brushes still aren't getting set to the right theme correctly. It might be because of how I styled some elements in the XAML, for example: `Foreground="{DynamicResource MicaWPF.Brushes.SystemAccentColorLight3}"`. I'll take a closer look at MicaWPF to see what's best practice.

Also, could you please keep WPF in the namespace? Thanks! 😊
One more thing—why did you refactor the namespaces to exclude the curly braces? I'm not too familiar with this in C#.

unchihugo commented:
Okay, it seems like it was an oversight by me—I shouldve used `AccentFillColorDefault` instead of `SystemAccentColorLight2` and `SystemAccentColorLight3`.

Could you please revert the namespace changes, and then I'll be ready to merge this PR. Thank you so much for all your hard work! I had some challenges figuring out the theme switching, so I really appreciate your help on this!

LiAuTraver commented:
hi, sorry for the inconvenience!
as for namespaces, my IDE(resharper) automatically does that when I format the code(it shows namespace does not correspond to their folder). I'll change it back.
> why did you refactor the namespaces to exclude the curly braces

sry, resharper does that😂
anyway,
```cs
namespace A;
class B{...}
// other stuff...
```
is equivalent to
```cs
namespace A{
        class B{...}
        // other stuff in current file...
}
```
personally I find this more readable and reduces nesting and I don't think it's a good idea to bring unnecesary braces back. Shall i do it anyway?
